[Miniflare] Error: could not determine executable to run

Environment:

OS: Ubuntu 22.04 LTS (WSL2)
Node 18.18.0 LTS
Npm: 10.2.3

---

Hi,

Trying to run miniflare on Windows WSL2 | Ubuntu 22.04.2 LTS, but am getting the following error.

```bash
$ npx miniflare

npm ERR! could not determine executable to run
```
Have tried both global  and dev dependency installation and the same issue occurs. The stack trace reports the following

```
36 verbose stack Error: could not determine executable to run
36 verbose stack     at getBinFromManifest (/home/sinclair/.nvm/versions/node/v18.18.0/lib/node_modules/npm/node_modules/libnpmexec/lib/get-bin-from-manifest.js:17:23)
36 verbose stack     at exec (/home/sinclair/.nvm/versions/node/v18.18.0/lib/node_modules/npm/node_modules/libnpmexec/lib/index.js:188:15)
36 verbose stack     at async module.exports (/home/sinclair/.nvm/versions/node/v18.18.0/lib/node_modules/npm/lib/cli-entry.js:61:5)
37 verbose pkgid miniflare@3.20231025.1
38 verbose cwd /home/sinclair
39 verbose Linux 5.10.102.1-microsoft-standard-WSL2
40 verbose node v18.18.0
41 verbose npm  v10.2.3
42 error could not determine executable to run
43 verbose exit 1
44 timing npm Completed in 183ms
45 verbose code 1
46 error A complete log of this run can be found in: /home/sinclair/.npm/_logs/2023-11-07T09_18_48_572Z-debug-0.log
```

Going back to `2.14.1` worked for me.  I am running on MacOS 13.6.

I ran into this issue, I tried the steps above, nothing.  I tried a number of [releases](https://github.com/cloudflare/miniflare/releases) on this page.

```bash
# this is the first version 3 release on that page
$ npx miniflare@3.20230904.0
npm ERR! could not determine executable to run
```

Stack trace the same as @sinclairzx81 except my own details.

Running miniflare@2.14.1

```bash
$ npx miniflare@2.14.1
[mf:inf] Worker reloaded! (95B)
[mf:inf] Listening on :8787
[mf:inf] - http://127.0.0.1:8787
[mf:inf] - http://192.168.0.244:8787
[mf:inf] - http://172.21.21.247:8787
GET / 200 OK (11.66ms)
GET /favicon.ico 200 OK (2.38ms)
```

I am running on:

MacOS 13.6
Node 18.18.2
NPM 9.8.1

--- a/.changeset/loud-taxis-drop.md
+++ b/.changeset/loud-taxis-drop.md
@@ -1,0 +1,7 @@
+---
+"miniflare": patch
+---
+
+fix: add `miniflare` `bin` entry
+
+Miniflare 3 doesn't include a CLI anymore, but should log a useful error stating this when running `npx miniflare`. We had a script for this, but it wasn't correctly hooked up. :facepalm: This change makes sure the required `bin` entry exists.

--- a/packages/miniflare/bootstrap.js
+++ b/packages/miniflare/bootstrap.js
@@ -1,3 +1,4 @@
+#!/usr/bin/env node
 const { Log } = require(".");
 
 const log = new Log();

--- a/packages/miniflare/package.json
+++ b/packages/miniflare/package.json
@@ -37,6 +37,9 @@
 		"lint:fix": "pnpm run lint -- --fix",
 		"types:build": "node scripts/types.mjs tsconfig.json && node scripts/types.mjs src/workers/tsconfig.json"
 	},
+	"bin": {
+		"miniflare": "bootstrap.js"
+	},
 	"dependencies": {
 		"acorn": "^8.8.0",
 		"acorn-walk": "^8.2.0",

